components/identifier-generator/front/src/feature/components/ScopeAndLocaleSelector.tsx

```diff
@@ -0,0 +1,68 @@
+import React, {useCallback, useMemo} from 'react';
+import {Channel, ChannelCode, LocaleCode, useTranslate} from '@akeneo-pim-community/shared';
+import {ScopeSelector} from './ScopeSelector';
+import {useGetAttributeByCode} from '../hooks/useGetAttributeByCode';
+import {Helper, Table} from 'akeneo-design-system';
+import {LocaleSelector} from './LocaleSelector';
+import {useGetScopes} from '../hooks';
+import {Styled} from './Styled';
+
+type Props = {
+  attributeCode: string;
+  locale?: LocaleCode | null;
+  scope?: ChannelCode | null;
+  onChange: ({scope, locale}: {scope?: string | null; locale?: string | null}) => void;
+};
+
+const ScopeAndLocaleSelector: React.FC<Props> = ({attributeCode, locale = null, scope = null, onChange}) => {
+  const translate = useTranslate();
+  const {data, isLoading, error} = useGetAttributeByCode(attributeCode);
+  const handleScopeChange = useCallback(
+    (scope: ChannelCode | null) => {
+      onChange({scope});
+    },
+    [onChange]
+  );
+  const {data: scopes} = useGetScopes();
+
+  const selectedScope = useMemo(
+    () => (data?.scopable ? scopes?.find(({code}: Channel) => code === scope) : undefined),
+    [data, scope, scopes]
+  );
+
+  const handleLocaleChange = useCallback(
+    (locale: LocaleCode | null) => {
+      onChange({locale});
+    },
+    [onChange]
+  );
```

Comment:
I don't understand what this function do.
Is this simpler ?

```suggestion
  const handleLocaleChange = useCallback(onChange),
    [onChange]
  );
```

Reply:
In this component, the values expected from the `onChange` function is `onChange({locale, scope})` so for each handleChange, we either set the value of `locale` or `scope`.